crates/bevy_input/src/gamepad.rs

```diff
@@ -1240,6 +1241,104 @@ const ALL_AXIS_TYPES: [GamepadAxisType; 6] = [
     GamepadAxisType::RightZ,
 ];
 
+/// The intensity at which a gamepad's force-feedback motors may rumble.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct GamepadRumbleIntensity {
+    /// The rumble intensity of the strong gamepad motor
+    ///
+    /// Ranges from 0.0 to 1.0
+    ///
+    /// By convention, this is usually a low-frequency motor on the left-hand
+    /// side of the gamepad, though it may vary across platforms and hardware.
+    pub strong: f32,
+    /// The rumble intensity of the weak gamepad motor
+    ///
+    /// Ranges from 0.0 to 1.0
+    ///
+    /// By convention, this is usually a high-frequency motor on the right-hand
+    /// side of the gamepad, though it may vary across platforms and hardware.
+    pub weak: f32,
+}
+
+impl GamepadRumbleIntensity {
+    /// Rumble both gamepad motors at maximum intensity
+    pub const MAX: Self = GamepadRumbleIntensity {
+        strong: 1.0,
+        weak: 1.0,
+    };
+
+    /// Rumble the weak motor at maximum intensity
+    pub const WEAK_MAX: Self = GamepadRumbleIntensity {
+        strong: 0.0,
+        weak: 1.0,
+    };
+
+    /// Rumble the strong motor at maximum intensity
+    pub const STRONG_MAX: Self = GamepadRumbleIntensity {
+        strong: 1.0,
+        weak: 0.0,
+    };
+
+    /// Creates a new rumble intensity with weak motor intensity set to the given value
+    pub fn weak(weak: f32) -> Self {
+        Self { weak, strong: 0.0 }
+    }
+
+    /// Creates a new rumble intensity with strong motor intensity set to the given value
+    pub fn strong(strong: f32) -> Self {
+        Self { strong, weak: 0.0 }
+    }
+}
+
+/// An event that controls force-feedback rumbling of a [`Gamepad`]
```

Comment:
nit doc: this needs a "vibration" somewhere. The feature has so many names, it's very easy to overlook if you are not using the right keyword.

Reply:
maybe a doc alias? https://doc.rust-lang.org/rustdoc/advanced-features.html#add-aliases-for-an-item-in-documentation-search